schemas/didDocument/didDocument.json

```diff
@@ -0,0 +1,33 @@
+{
+    "$id": "/did-core.didDocument",
+    "type": "object",
+    "properties": {
+      "@context": {
+        "title": "@context",
+        "description": "https://w3c.github.io/did-core-registry/#context",
+        "anyOf": [
+          { "type": "string" },
+          { "type": "array" },
```

Comment:
Spec says that `@context` should only allow an array where the first member in the array is the DID Core Context URL.

Reply:
What happens when https://www.w3.org/ns/did/v1 becomes https://www.w3.org/ns/did/v2...?

Reply:
You can test revisions here: https://www.jsonschemavalidator.net/

the json schema now enforces the spec language.

Reply:
> What happens when https://www.w3.org/ns/did/v1 becomes https://www.w3.org/ns/did/v2...?

The (new) spec will state that the first member in the array MUST be https://www.w3.org/ns/did/v2

Note that this doesn't mean that old implementations will stop working, because the old v1 context will continue to exist and work (nothing breaks). There will just be a new `@context` for implementations that want to be v2 compliant. This is effectively how we do backwards-compatible versioning in JSON-LD.

It does raise a question wrt. the use of JSON Schema... I don't remember if the newest version of JSON Schema allows you to do variadic loading of alternate schemas ... so if you see v1, load schemas 1, 2, and 3... and if you see v2, load schemas 4, 5, and 6. I expect that if that's not supported that there will have to eventually be two types of JSON Schemas (for JSON-LD -- but only if we want to be able to deprecate features, which I expect we will)... the JSON-only folks won't be able to deprecate features, I expect.

There's a lot of thinking in this area wrt. the use of JSON Schema that hasn't been done... I don't forsee any terrible things at this point, but we will need to establish best practices. Keep in mind that JSON Schema only works for JSON, so it's still an open question if it applies to the CBOR version.

Reply:
As I understood the F2F outcome regarding CBOR, they would not be using features that break with JSON compatibility... so a JSON only data model with json schema, should be sufficient for all representations.

Reply:
We can easily version the entire JSON Schema Schema Tree, using URIs if needed.

Reply:
> As I understood the F2F outcome regarding CBOR, they would not be using features that break with JSON compatibility (like tags)... so a JSON only data model with json schema

What was agreed to at the F2F was the ability to loss-lessly translate between representations. We didn't agree that you should be able to apply JSON Schema to a CBOR data structure that hasn't gone through a spec-defined translation to JSON. Yes, we could just say "follow the standard CBOR->JSON conversion process", but I expect CBOR-only die hards to be annoyed by that (and rightly so). I believe we should be able to convert a CBOR representation of the DID Core data model to a JSON representation and then run JSON Schema over that representation and have no surprises, I do think there is consensus there. 

The issues I foresee are:

* CBOR representations could be far more compact if we use integers as keys instead of strings... this is a pretty typical CBOR thing to do. This applies to values such as Ed25519Signature2019
* It is possible to use CBOR tags as long as the conversion process from tagged CBOR to DID Core data model is documented in the DID Core spec (or other relevant specification).
* CBOR-LD, if we ever get around to writing that spec, will be capable of automatically and deterministically compressing term/class URLs to 16-bit, 32-bit, 64-bit, and 128-bit values. If we do this right, the target conversion values will be the same across all languages. We should be careful to not prevent us from doing this optimization in the future because the storage savings can be significant for large VCs and DID Documents.

Just trying to put that in your head so it's there as you puzzle through the best approach for the DID Core Registry.

Reply:
So if we document every property in the did core registry, and someone adds the relevant CBOR definitions, sounds like we will be good.

I have created #5  and #6 to track adding JSON-LD and CBOR definitions for `id`... they are blocked by the "where to host JSON-LD contexts" question.

Reply:
> So if we document every property in the did core registry, and someone adds the relevant CBOR definitions, sounds like we will be good.

Yes, and we should be able to auto-generate the CBOR identifiers from the JSON-LD vocabulary terms (which are required for anything to be added to the DID Core registry).